src/components/guide/competency-select.js

```diff
@@ -0,0 +1,44 @@
+import PropTypes from "prop-types";
+import {Component} from "react";
+import style from "./style";
+
+export default class CompetencySelect extends Component {
+  static propTypes = {
+    competencies: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired
+      })
+    ).isRequired,
+    displayedCompetency: PropTypes.shape({
+      id: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired
+    }).isRequired,
+    displayCompetency: PropTypes.func.isRequired,
+    tabBadge: PropTypes.func.isRequired
+  }
+  displayCompetency = (e) => {
+    this.props.displayCompetency(e.target.value);
+  }
+  render() {
+    const {competencies, displayedCompetency, tabBadge} = this.props;
+
+    return (
+      <div className={style.competencySelect}>
+        <select
+          value={displayedCompetency.id}
+          className={style.mobileSelect}
+          onChange={this.displayCompetency}
```

Comment:
could you access from displayCompetency from props and use an arrow function to pass back the event instead of making this a part of the class?

const {displayCompetency} = this.props;
onChange={(e) => displayCompetency(e.target.value)}

Reply:
I don't think we have it on, but with more strict rules they warn against creating new functions in handlers. I think the argument is that every time it renders and has to create and bind a new function, whereas having it on the class object, it only creates the function once